qualys-fim/operations.py

```diff
@@ -0,0 +1,160 @@
+""" 
+Copyright start 
+Copyright (C) 2008 - 2021 Fortinet Inc. 
+All rights reserved. 
+FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE 
+Copyright end 
+"""
+
+import requests
+
+from connectors.core.connector import get_logger, ConnectorError
+
+logger = get_logger('qualys-fim')
+
+
+class QualysFIM(object):
+    def __init__(self, config):
+        self.server_url = config.get('server_url')
+        if not self.server_url.startswith('https://'):
+            self.server_url = 'https://' + self.server_url
+        if not self.server_url.endswith('/'):
+            self.server_url += '/'
+        self.username = config.get('username')
+        self.password = config.get('password')
+        self.verify_ssl = config.get('verify_ssl')
+
+    def make_request(self, endpoint=None, method='GET', data=None, params=None, files=None, headers=None):
+        try:
+            url = self.server_url + endpoint
+            response = requests.request(method, url, params=params, files=files, data=data, headers=headers,
+                                        verify=self.verify_ssl)
+            if response.status_code == 201 or response.status_code == 200:
+                return response.json()
+            else:
+                logger.error(response.text)
+                raise ConnectorError({'status_code': response.status_code, 'message': response.reason})
+        except requests.exceptions.SSLError:
+            raise ConnectorError('SSL certificate validation failed')
+        except requests.exceptions.ConnectTimeout:
+            raise ConnectorError('The request timed out while trying to connect to the server')
+        except requests.exceptions.ReadTimeout:
+            raise ConnectorError('The server did not send any data in the allotted amount of time')
+        except requests.exceptions.ConnectionError:
+            raise ConnectorError('Invalid endpoint or credentials')
+        except Exception as err:
+            logger.exception(str(err))
+            raise ConnectorError(str(err))
+
+
+def generate_authentication_token(config):
+    try:
+        endpoint = config.get('server_url') + '/auth'
+        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
+        data = {'username': config.get('username'), 'password': config.get('password'), 'token': True}
+        token = requests.post(url=endpoint, headers=headers, data=data).text
```

Comment:
need to handle the error conditions and successful conditions. also you can use the make_request action to handle t better way.